### PR TITLE
js(analytics): update pinpoint service provider configuration options for V6

### PR DESCRIPTION
**Pull request by 5d**
#### Description of changes:
- update the doc for pinpoint service provider configuration options
  
#### Related GitHub issue #, if available:
https://github.com/aws-amplify/amplify-js/pull/12521

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [X] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [X] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib/analytics/js/existing-resources.mdx [hunk 1]
```diff
@@ -8,11 +8,25 @@ import { Amplify } from 'aws-amplify';
...
       appId: 'XXXXXXXXXXabcdefghij1234567890ab',
+
       // REQUIRED -  Amazon service region
-      region: 'us-east-1'
+      region: 'us-east-1',
+
+      // OPTIONAL - The buffer size for events in number of items.
+      bufferSize: 1000,
+
+      // OPTIONAL - The number of events to be deleted from the buffer when flushed.
```
Comment (jimblanc) [inline src/fragments/lib/analytics/js/existing-resources.mdx:22]:
```suggestion
      // OPTIONAL - How many events will be flushed from the buffer per batch.
```

File: src/fragments/lib/analytics/js/existing-resources.mdx [hunk 2]
```diff
@@ -8,11 +8,25 @@ import { Amplify } from 'aws-amplify';
...
+    ...Amplify.getConfig().Analytics,
     Pinpoint: {
       // REQUIRED -  Amazon Pinpoint App Client ID
       appId: 'XXXXXXXXXXabcdefghij1234567890ab',
+
       // REQUIRED -  Amazon service region
-      region: 'us-east-1'
+      region: 'us-east-1',
+
+      // OPTIONAL - The buffer size for events in number of items.
```
Comment (jimblanc) [inline src/fragments/lib/analytics/js/existing-resources.mdx:19]:
```suggestion
      // OPTIONAL - How many events can be buffered at once.
```

**Review by jimblanc — APPROVED** [review]
LGTM, some non-blocking nits

**Review by renebrandel — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]

File: src/fragments/lib/analytics/js/existing-resources.mdx (hunk 2 above)
Comment (5d) [reply src/fragments/lib/analytics/js/existing-resources.mdx]: The other Analytics service providers are using the same description for these options. I can update all of them at once in another PR. 

ref: https://github.com/aws-amplify/docs/blob/next-release/main/src/fragments/lib/analytics/js/streaming.mdx#L35-L45

**Review by AllanZhengYP — APPROVED** [review]

**Pull request merged**